Lib/_pylong.py

```diff
@@ -219,7 +308,10 @@ def int_from_string(s):
     # and underscores, and stripped leading whitespace.  The input can still
     # contain underscores and have trailing whitespace.
     s = s.rstrip().replace('_', '')
-    return _str_to_int_inner(s)
+    func = _str_to_int_inner
+    if len(s) >= 3_500_000 and _decimal is not None:
```

Comment:
On my computer I have equilibrium around 3_000_000. For 3_500_000 the new code is 20% faster, for 2_500_000 it is 20% slower.

So it is very close.

Reply:
I see breakeven closer to 3.3.M, so 3.5M is a clear winner for both of us. "Do no harm" is more important to me than "aggressively pursue relatively small speedups" here, so I'll leave it alone.